[dynamo] Refactor into torch/_inductor/runtime/compile_tasks.py

jansel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #124681
* #124592



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

Differential Revision: [D56723769](https://our.internmc.facebook.com/intern/diff/D56723769)

jansel committed:
Update

[ghstack-poisoned]

jansel committed:
Update

[ghstack-poisoned]

masnesral commented on torch/_inductor/runtime/compile_tasks.py:
```diff
@@ -0,0 +1,68 @@
+
+def _reload_python_module_in_subproc(key, path):
+    codecache = sys.modules.get("torch._inductor.codecache")
+    if codecache:
+        return codecache.PyCodeCache.load_by_key_path(key, path)
+    else:
+        return _reload_python_module(key, path)
```
Nit: AFAICT, this PR is mostly just moving methods around. This is the one place I don't quite grok. What's happening that requires the sys.modules.get? So a comment about what's going on here would be nice.

jansel replied:
It is so that in the next PR when we do:
```
sys.modules["torch"] = MockTorch()
```
we don't try to go via codecache.

masnesral reviewed (approved):
One nit, but LGTM

jansel commented:
@pytorchbot merge

jansel commented:
@pytorchbot merge

jeanschmidt commented:
@pytorchbot revert -m "breaking internal tests, check D56522594 -c nosignal

jeanschmidt commented:
@pytorchbot revert -m "breaking internal tests, check D56522594" -c nosignal

masnesral committed:
Update on "[dynamo] Refactor into torch/_inductor/runtime/compile_tasks.py"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

masnesral commented:
@masnesral has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56723769).

masnesral committed:
rebase past broken benchmark on "[dynamo] Refactor into torch/_inductor/runtime/compile_tasks.py"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

Differential Revision: [D56723769](https://our.internmc.facebook.com/intern/diff/D56723769)

[ghstack-poisoned]

masnesral commented:
@masnesral has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56723769).

masnesral commented:
@pytorchbot merge
